### PR TITLE
TSE Review for User interface Branch

### PR DESCRIPTION
**Pull request by Calvinoea**
Hello,

We (Hillario and I) are making a request for your review of our user interface milestone 2 submission as part of the Ruby curriculum administered by Microverse. 

The features include:

- Display prompts for users (hint: puts/print)
- Collection of user input 
- Basic mock flow of the game

Thanks for your review!

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice one :+1: 
Your game logic is almost done. We need to fix a few things before it's ready for implementation.
- [ ] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
An example of a variable terminated while loop.
```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```
- [ ] add logic for draw
- [ ] enable stickler in this repo and make sure all test cases pass

### Martin Afani

File: bin/main [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+# board_positions = ["", "", "", "", "", "", "", "", "",]
+
+board_display = []
+
+new_game = "-#{board_display[0]}-|-#{board_display[1]}-|-#{board_display[2]}-"
+
+puts new_game
+puts new_game
+puts new_game
```
Comment (whiz25) [inline bin/main]: - [ ] why three of these?

File: bin/main [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+if player_one_turn != 'x'
+  puts 'You did not enter X'
+else
+  board_display.push(player_one_turn)
+  # puts boardDisplay
+  puts new_game
+  puts 'Your turn, player two'
+  player_two_turn = gets.chomp.downcase
+  puts 'You did not enter O' if player_two_turn != 'o'
+end
```
Comment (whiz25) [inline bin/main]: - [ ] you need a loop to continuously repeat this step until there is a winner or a draw

File: bin/main [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+puts 'You win, player one'
```
Comment (whiz25) [inline bin/main]: - [ ] when does someone win?

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice Progress :+1: 
You have made some changes as suggested. But we need to fix a few things before we can merge this PR.

- [ ] check my comments for suggestions

### Martin Afani

File: bin/main.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+maximum_moves = 5
+
+loop do
+  Main.enter_position('Player_name')
+  Main.display_board
+  break if maximum_moves == 1
+
+  maximum_moves -= 1
+end
```
Comment (whiz25) [inline bin/main.rb]: - use a variable to terminate the loop.
- this should not be complex. you can even use just comments to show you thought process.
- for example,
```ruby
game_on = true
while game_on
# let players take turn until there is a winner
if winner
# tell which player won and end game
end
game_on = false
end
```

**Review by majovanilla — APPROVED** [review]
Hi team!!

Good job :heavy_check_mark: implementing the required changes!! :star: 

Your project is ready for the next stage!! :100:

### Milestone 2 approved. Please merge and continue.


If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

**Pull request merged**